pkg/skaffold/initializer/init_test.go

```diff
@@ -59,3 +61,120 @@ func TestPrintAnalyzeJSON(t *testing.T) {
 		})
 	}
 }
+
+func TestWalk(t *testing.T) {
+	emptyFile := ""
+	tests := []struct {
+		name                string
+		filesWithContents   map[string]string
+		expectedConfigs     []string
+		expectedDockerfiles []string
+		force               bool
+		shouldErr           bool
+	}{
+		{
+			name: "should return correct k8 configs and dockerfiles",
+			filesWithContents: map[string]string{
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+			},
+			force: false,
+			expectedConfigs: []string{
+				"config/test.yaml",
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+				"deploy/Dockerfile",
+			},
+			shouldErr: false,
+		},
+		{
+			name: "should skip hidden dir",
+			filesWithContents: map[string]string{
+				".hidden/test.yaml":  emptyFile,
+				"k8pod.yml":          emptyFile,
+				"README":             emptyFile,
+				".hidden/Dockerfile": emptyFile,
+				"Dockerfile":         emptyFile,
+			},
+			force: false,
+			expectedConfigs: []string{
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+			},
+			shouldErr: false,
+		},
+		{
+			name: "should not error when skaffold.config present and force = true",
+			filesWithContents: map[string]string{
+				"skaffold.yaml": `apiVersion: skaffold/v1beta6
+kind: Config
+deploy:
+  kustomize: {}`,
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+			},
+			force: true,
+			expectedConfigs: []string{
+				"config/test.yaml",
+				"k8pod.yml",
+			},
+			expectedDockerfiles: []string{
+				"Dockerfile",
+				"deploy/Dockerfile",
+			},
+			shouldErr: false,
+		},
+		{
+			name: "should  error when skaffold.config present and force = false",
+			filesWithContents: map[string]string{
+				"config/test.yaml":  emptyFile,
+				"k8pod.yml":         emptyFile,
+				"README":            emptyFile,
+				"deploy/Dockerfile": emptyFile,
+				"Dockerfile":        emptyFile,
+				"skaffold.yaml": `apiVersion: skaffold/v1beta6
+kind: Config
+deploy:
+  kustomize: {}`,
+			},
+			force:               false,
+			expectedConfigs:     nil,
+			expectedDockerfiles: nil,
+			shouldErr:           true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			testDir, deleteFunc := testutil.NewTempDir(t)
+			rootDir := testDir.Root()
+			deleteFunc()
```

Comment:
This should be `defer deleteFunc()` and be moved one line higher

Reply:
We also usually don't use a `Func` suffix

Reply:
will fix this.